File: api/suggestions.php
<?php

/**
 * Suggestions API Handler
 * Handles all suggestion-related API endpoints
 */

require_once __DIR__ . '/config.php';
require_once __DIR__ . '/db.php';

setCorsHeaders();

$method = $_SERVER['REQUEST_METHOD'];
$id = $_GET['id'] ?? null;

// Sample suggestions data
$sampleSuggestions = [
    [
        'id' => 1,
        'name' => 'Mary Wanjiku',
        'email' => 'mary@email.com',
        'category' => 'school',
        'subject' => 'Add more schools in Northern Uganda',
        'message' => 'Please consider adding more schools from Gulu, Lira, and other northern regions.',
        'status' => 'pending',
        'created_at' => '2026-01-15'
    ],
    [
        'id' => 2,
        'name' => 'James Odhiambo',
        'email' => 'james@email.com',
        'category' => 'bursary',
        'subject' => 'More scholarship information',
        'message' => 'It would be helpful to have more details about scholarship requirements.',
        'status' => 'reviewed',
        'created_at' => '2026-02-10'
    ]
];

// Route requests
if ($method === 'GET' && $id) {
    getSuggestion($id);
} elseif ($method === 'GET') {
    getSuggestions();
} elseif ($method === 'POST') {
    submitSuggestion();
} else {
    jsonResponse(['error' => 'Method not allowed'], 405);
}

function getSuggestions()
{
    global $sampleSuggestions;

    $db = Database::getConnection();
    if ($db) {
        $suggestions = getAll('suggestions');
        if (count($suggestions) > 0) {
            jsonResponse([
                'success' => true,
                'suggestions' => $suggestions,
                'total' => count($suggestions)
            ]);
        }
    }

    jsonResponse([
        'success' => true,
        'suggestions' => $sampleSuggestions,
        'total' => count($sampleSuggestions)
    ]);
}

function getSuggestion($id)
{
    global $sampleSuggestions;

    foreach ($sampleSuggestions as $s) {
        if ($s['id'] == $id) {
            jsonResponse([
                'success' => true,
                'suggestion' => $s
            ]);
        }
    }

    $db = Database::getConnection();
    if ($db) {
        $suggestion = getById('suggestions', $id);
        if ($suggestion) {
            jsonResponse([
                'success' => true,
                'suggestion' => $suggestion
            ]);
        }
    }

    jsonResponse(['error' => 'Suggestion not found'], 404);
}

function submitSuggestion()
{
    $data = getJsonInput();

    if (empty($data['name']) || empty($data['message'])) {
        jsonResponse(['error' => 'Name and message are required'], 400);
    }

    $db = Database::getConnection();

    if ($db) {
        $data['created_at'] = date('Y-m-d H:i:s');
        $data['status'] = 'pending';

        $id = insert('suggestions', $data);

        if ($id) {
            jsonResponse([
                'success' => true,
                'message' => 'Suggestion submitted successfully',
                'id' => $id
            ], 201);
        }
    }

    jsonResponse([
        'success' => true,
        'message' => 'Suggestion submitted (demo mode)'
    ], 201);
}

File: api/suggestions.py
"""
Suggestions API Handler for Vercel
Handles all suggestion-related API endpoints
"""

import json

def get_response(status_code, body):
    return {
        "statusCode": status_code,
        "headers": {
            "Content-Type": "application/json",
            "Access-Control-Allow-Origin": "*",
            "Access-Control-Allow-Methods": "GET, POST, PUT, DELETE, OPTIONS",
            "Access-Control-Allow-Headers": "Content-Type, Authorization"
        },
        "body": json.dumps(body)
    }

SAMPLE_SUGGESTIONS = [
    {
        "id": 1,
        "name": "Mary Wanjiku",
        "email": "mary@email.com",
        "category": "school",
        "subject": "Add more schools in Northern Uganda",
        "message": "Please consider adding more schools from Gulu, Lira, and other northern regions.",
        "status": "pending",
        "created_at": "2026-01-15"
    },
    {
        "id": 2,
        "name": "James Odhiambo",
        "email": "james@email.com",
        "category": "bursary",
        "subject": "More scholarship information",
        "message": "It would be helpful to have more details about scholarship requirements.",
        "status": "reviewed",
        "created_at": "2026-02-10"
    }
]

def handle(req, resource_id=None, params=None):
    method = req.get('method', 'GET')
    
    if method == 'OPTIONS':
        return get_response(200, {"message": "OK"})
    
    if method == 'GET' and resource_id is None:
        return get_response(200, {
            "success": True,
            "suggestions": SAMPLE_SUGGESTIONS,
            "total": len(SAMPLE_SUGGESTIONS)
        })
    
    if method == 'GET' and resource_id:
        suggestion = next((s for s in SAMPLE_SUGGESTIONS if str(s["id"]) == resource_id), None)
        if suggestion:
            return get_response(200, {"success": True, "suggestion": suggestion})
        return get_response(404, {"error": "Suggestion not found"})
    
    if method == 'POST':
        return get_response(201, {"success": True, "message": "Suggestion submitted successfully"})
    
    return get_response(405, {"error": "Method not allowed"})

